### PR TITLE
Add logging for a file path that can't be cleaned

### PR DESCRIPTION
**Pull request by emilieschario**
Closes https://github.com/fishtown-analytics/dbt/issues/2059

In a case where your `dbt_project.yml` file has:
```
source-paths: ["models"]

clean-targets:
  - "models"
  - "dbt_modules"
```

when you run `dbt clean` you'll see a message that looks like:

```
Checking models/*
Checking dbt_modules/*
 Cleaned dbt_modules/*
```

Because it doesn't tell you you did something wrong, you may not notice that the `model/*` file wasn't cleaned (rightfully so!). This MR proposes that the messaging be:
```
Checking model/*
model/* cannot be cleaned.
Checking dbt_modules/*
 Cleaned dbt_modules/*
```

No functional differences, just more details.

### CODE REVIEW

**Comment by emilieschario** [comment]
@drewbanin I think these tests are failing because someone internal has to kick them off

**Comment by drewbanin** [comment]
Thanks for making this PR @emilieschario! I just kicked off the tests here.

This is super minor, but what's your rationale behind not indenting the log line that you added here? And what would you think about updating this line to be _even more_ apparent with something like:
```
 ERROR: not cleaning {}/* because it is protected
```

Let me know what you think!

**Comment by emilieschario** [comment]
On the not indenting piece, it was being it was hard to anticipate the length, so it could be a short name like `model` or it could be `kjdhfskjfdlakjldkj` so
```
Checking model/*
kjdhfskjfdlakjldkj/* cannot be cleaned.
Checking dbt_modules/*
 Cleaned dbt_modules/*
```

But I love love love the error message! I'm going to update now so that you can kick off tests again, if you're still around

File: core/dbt/task/clean.py [hunk 1]
```diff
@@ -35,4 +35,6 @@ def run(self):
             if not self.__is_protected_path(path):
                 shutil.rmtree(path, True)
                 logger.info(" Cleaned {}/*".format(path))
+            else:
+                logger.info("{}/* cannot be cleaned".format(path))
```
Comment (emilieschario) [inline core/dbt/task/clean.py]:
```suggestion
                logger.info("ERROR: not cleaning {}/* because it is protected".format(path))
```

**Comment by emilieschario** [comment]
Updated @drewbanin!

**Comment by drewbanin** [comment]
Thanks @emilieschario! This LGTM, but I have terrible news: your new change has a line that's longer than 80 characters, so it won't pass our code formatting CI check.

Can you try changing this line:
```
                logger.info("ERROR: not cleaning {}/* because it is protected".format(path))
```
to look like:
```
                logger.info("ERROR: not cleaning {}/* because it is "
                            "protected".format(path))
```

This will fix the pep8 error that might show up in Circle momentarily :)

**Comment by emilieschario** [comment]
Thanks for your patience here @drewbanin! Updated!

**Comment by drewbanin** [comment]
np! Thanks for that - just kicked off the tests again and will get this merge when they pass :)

**Comment by drewbanin** [comment]
hey @emilieschario - this code is definitely read to roll, but in testing this feature, I did notice a big problem!

Check out this code (unrelated to your PR): https://github.com/fishtown-analytics/dbt/pull/2060/files#diff-0927dc49a0a3b3ad824b956c8f0abb3fR21-R26

On line 24 here, we call:
```
protected_abs_paths = [os.path.abspath for p in protected_paths]
```

The problem is that we're not _calling_ the `abspath` function, so `protected_abs_paths` takes on a value like:
```
> protected_abs_paths
[<function abspath at 0x101f3e950>, <function abspath at 0x101f3e950>, <function abspath at 0x101f3e950>]
```

These list elements are pointers to the `abspath` method, not absolute paths themselves! As a result, the `__is_protected_path` method does not actually check if paths are protected :/

If you're interested in fixing that issue in the scope of this PR, I think that would be a really good and helpful change. If not, we can totally queue up a second ticket to address it.

Let me know!=

**Comment by emilieschario** [comment]
woah. so why didn't my file get cleaned before if it wasn't because it was being deemed a file path? how did I come to this conclusion?  😕 

I think I got it. Thanks for giving me the chance to do that.

**Comment by emilieschario** [comment]
Thanks for the opportunity @drewbanin!

**Pull request merged**